pkg/cfn/manager/api.go

```diff
@@ -79,17 +95,25 @@ func NewStackCollection(provider api.ClusterProvider, spec *api.ClusterConfig) *
 		tags = append(tags, newTag(key, value))
 	}
 	return &StackCollection{
-		provider:   provider,
-		spec:       spec,
-		sharedTags: tags,
+		spec:              spec,
+		sharedTags:        tags,
+		cloudformationAPI: provider.CloudFormation(),
+		ec2API:            provider.EC2(),
+		eksAPI:            provider.EKS(),
+		iamAPI:            provider.IAM(),
+		cloudTrailAPI:     provider.CloudTrail(),
+		disableRollback:   provider.CloudFormationDisableRollback(),
+		roleARN:           provider.CloudFormationRoleARN(),
+		region:            provider.Region(),
+		waitTimeout:       provider.WaitTimeout(),
```

Comment:
I really don't think this is an improvement, I've made the change more to illustrate how crazy the provider usage is inside the manager package. 

I'm open for reverting it and trying to address the usage of ctl inside the manager package in a separate PR. LMK if anyone disagrees

Reply:
i am really torn. on the one hand, this will make us do something about the massive manager package sooner rather than later. on the other, i hate it. and on a third hand, we _don't_ deal with it soon and someone can easily look at this and say "hmm we should have a thing which wraps all these APIs", which is probably how the provider began to begin with before people started throwing all the other junk in there.

so in conclusion: idk